bindings/nodejs/src/wallet.rs

```diff
@@ -16,26 +19,54 @@ use crate::{
     build_js_error, client::ClientMethodHandler, destroyed_err, secret_manager::SecretManagerMethodHandler, NodejsError,
 };
 
-pub type WalletMethodHandler = Arc<RwLock<Option<Wallet>>>;
+pub struct WalletMethodHandlerInner(Option<Wallet>);
+
+impl Drop for WalletMethodHandlerInner {
+    fn drop(&mut self) {
+        log::debug!("drop WalletMethodHandlerInner");
+        // Request to stop the background syncing silently if this wallet hasn't been destroyed yet
+        if let Some(wallet) = self.0.take() {
```

Comment:
So, if `destroy_wallet` is called before dropping, this will not stop the syncing. Perhaps we should repeat this code in that fn?

Reply:
> @marc2332 Just making sure you've seen this question :)

Nice timing :joy:

Reply:
Does that answer your question @DaughterOfMars ?

Reply:
That seems to be a different question. I'm saying that this check will do nothing in the Drop impl if destroy_wallet was called first. So if you do this order of operations:
1. Start background syncing
2. Call `destroyWallet`
3. Drop the memory

the background syncing will still continue.

Reply:
> That seems to be a different question. I'm saying that this check will do nothing in the Drop impl if destroy_wallet was called first. So if you do this order of operations:
> 
>     1. Start background syncing
> 
>     2. Call `destroyWallet`
> 
>     3. Drop the memory
> 
> 
> the background syncing will still continue.

That's exactly what I am talking about, I could also make it call `request_stop_background_syncing` in `destroyWallet` so it stops the background syncing